libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationPresenter.java

```diff
@@ -69,9 +70,12 @@ void onNavigationLocationUpdate(Location location) {
 
   public void onInstructionListVisibilityChanged(boolean visible) {
     if (visible) {
+      recenterButtonVisible = view.isRecenterButtonVisible();
       view.hideRecenterBtn();
     } else {
-      view.showRecenterBtn();
+      if (recenterButtonVisible) {
```

Comment:
I think `recenterButtonVisible` can be eliminated by checking if the `SummaryBottomSheet` is visible here instead with `view.isSummaryBottomSheetHidden()`.  We only want to show if the bottomsheet is currently hidden right?